From a96b79f909403b53ea4c577912bdf611da9ff064 Mon Sep 17 00:00:00 2001
From: Jeff Gilbert <jgilbert@mozilla.com>
Date: Fri, 26 Jul 2019 17:33:16 -0700
Subject: [PATCH] Explicitly instantiate TIntermTraverser::traverse(TIntermNode *).

(on behalf of RewriteAtomicFunctionExpressions)
The compile will occasionally choose to inline all instances of this
function, and therefore not export this entrypoint into the object file,
leaving other TUs (RewriteAtomicFunctionExpressions) with missing
link symbols.

Bug: angleproject:3752
Change-Id: Ida778bcd897a85567680911a88aae88a3797aa93
Reviewed-on: https://chromium-review.googlesource.com/c/angle/angle/+/1721954
Commit-Queue: Jeff Gilbert <jgilbert@mozilla.com>
Commit-Queue: Jamie Madill <jmadill@chromium.org>
Reviewed-by: Jamie Madill <jmadill@chromium.org>
---

diff --git a/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp b/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
index c38baa1..b587090 100644
--- a/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
+++ b/gfx/angle/checkout/src/compiler/translator/tree_util/IntermTraverse.cpp
@@ -50,6 +50,10 @@
     }
 }
 
+// Instantiate template for RewriteAtomicFunctionExpressions, in case this gets inlined thus not
+// exported from the TU.
+template void TIntermTraverser::traverse(TIntermNode *);
+
 void TIntermNode::traverse(TIntermTraverser *it)
 {
     it->traverse(this);
